gonzo/clouds/compute.py

```diff
@@ -126,26 +126,41 @@ def create_instance(self, image_name, name, owner, user_data=None,
         if security_groups is None:
             security_groups = []
 
-        for security_group in security_groups:
-            self.create_if_not_exist_security_group(security_group)
+        for index, security_group in enumerate(security_groups):
+            self.create_if_not_exist_security_group(security_group, subnet_id)
+            security_groups[index] = self.get_security_group(
+                security_group, subnet_id
+            )
 
         security_groups_for_launch = self.security_groups_for_launch(
             security_groups)
 
-        # Availability Zone
```

Comment:
i'd keep the comment explaining what `az` is because i wouldn't have understood otherwise.